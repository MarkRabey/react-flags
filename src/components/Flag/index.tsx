import React, {useState, useEffect} from 'react';
import {Text, View, Image} from 'react-native';

import images from '../../images';
interface Props {
  countryCode: string;
  size?: number;
  style?: any;
  variant?: 'emoji' | 'flat' | '3d';
}

const Flag = ({countryCode, size = 16, style, variant = 'emoji'}: Props) => {
  const [flag, setFlag] = useState<string | null>(null);
  useEffect(() => {
    if (variant === 'emoji') {
      const codePoints = countryCode
        .toUpperCase()
        .split('')
        .map(char => 127397 + char.charCodeAt(0));
      setFlag(String.fromCodePoint(...codePoints));
    } else {
      setFlag(countryCode.toLowerCase());
    }
  }, [countryCode, variant]);

  const imageScale = 0.7;

  return (
    <View style={{...style}}>
      {variant === 'emoji' ? (
        <Text style={{fontSize: size}}>{flag}</Text>
      ) : (
        <Image
          source={images[variant][flag]}
          style={{width: size * imageScale * 1.6, height: size * imageScale}}
        />
      )}
    </View>
  );
};

export default Flag;
