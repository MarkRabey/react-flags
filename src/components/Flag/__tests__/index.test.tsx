import React from 'react';
import renderer from 'react-test-renderer';

import Flag from '../index';

describe('Flag', () => {
  test('renders correctly', () => {
    const result = renderer.create(<Flag countryCode="CA" />).toJSON();
    expect(result).toMatchSnapshot();
  });

  test('renders flat image correctly', () => {
    const result = renderer
      .create(<Flag countryCode="CA" variant="flat" />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  test('checks country code for "IN"', () => {
    const result = renderer
      .create(<Flag countryCode="IN" variant="flat" />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  test('checks country code for "DO"', () => {
    const result = renderer
      .create(<Flag countryCode="DO" variant="flat" />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
