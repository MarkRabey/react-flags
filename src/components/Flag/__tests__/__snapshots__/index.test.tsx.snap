// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  />
</View>
`;
